src/librustdoc/html/render/write_shared.rs

```diff
@@ -247,64 +210,83 @@ pub(super) fn write_shared(
         themes.insert(theme.to_owned());
     }
 
-    if (*cx.shared).layout.logo.is_empty() {
-        write_toolchain("rust-logo.png", static_files::RUST_LOGO)?;
-    }
-    if (*cx.shared).layout.favicon.is_empty() {
-        write_toolchain("favicon.svg", static_files::RUST_FAVICON_SVG)?;
-        write_toolchain("favicon-16x16.png", static_files::RUST_FAVICON_PNG_16)?;
-        write_toolchain("favicon-32x32.png", static_files::RUST_FAVICON_PNG_32)?;
-    }
-    write_toolchain("brush.svg", static_files::BRUSH_SVG)?;
-    write_toolchain("wheel.svg", static_files::WHEEL_SVG)?;
-    write_toolchain("clipboard.svg", static_files::CLIPBOARD_SVG)?;
-    write_toolchain("down-arrow.svg", static_files::DOWN_ARROW_SVG)?;
-    write_toolchain("toggle-minus.svg", static_files::TOGGLE_MINUS_PNG)?;
-    write_toolchain("toggle-plus.svg", static_files::TOGGLE_PLUS_PNG)?;
-
     let mut themes: Vec<&String> = themes.iter().collect();
     themes.sort();
 
+    // Given "foo.svg", return e.g. "url(\"foo1.58.0.svg\")"
+    fn ver_url(cx: &Context<'_>, basename: &'static str) -> String {
+        format!(
+            "url(\"{}\")",
+            SharedResource::ToolchainSpecific { basename }
+                .path(cx)
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap()
+        )
+    }
+
+    // We use the AUTOREPLACE mechanism to inject into our static JS and CSS certain
+    // values that are only known at doc build time. Since this mechanism is somewhat
+    // surprising when reading the code, please limit it to rustdoc.css and storage.js.
+    write_minify(
+        "rustdoc.css",
+        static_files::RUSTDOC_CSS
+            .replace(
+                "/* AUTOREPLACE: */url(\"toggle-minus.svg\")",
+                &ver_url(cx, "toggle-minus.svg"),
+            )
+            .replace("/* AUTOREPLACE: */url(\"toggle-plus.svg\")", &ver_url(cx, "toggle-plus.svg"))
+            .replace("/* AUTOREPLACE: */url(\"down-arrow.svg\")", &ver_url(cx, "down-arrow.svg")),
+        cx,
+        options,
+    )?;
+
     // FIXME: this should probably not be a toolchain file since it depends on `--theme`.
     // But it seems a shame to copy it over and over when it's almost always the same.
     // Maybe we can change the representation to move this out of main.js?
     write_minify(
```

Comment:
Hmm, I think this should be using `cx.write_minify(InvocationSpecific)`, since it can vary from crate to crate. Unfortunately we don't have a good way of testing this before it hits nightly right now.

Reply:
The only thing that is invocation specific is the list of themes, right? What if I put those in a `data-` attribute in the HTML instead? The list is quite short, so duplication is not a big concern.

Reply:
That would be great! Thanks :)